src/blockmetadata.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "blockmetadata.h"
+#include "exceptions.h"
+#include "log.h"
+#include "util/serialize.h"
+#include <sstream>
+
+/*
+	BlockMetadata
+*/
+void BlockMetadata::serialize(std::ostream &os, u8 blockver, bool disk) const
+{
+	u8 version = blockver >= 29 ? 1 : 0; // always 1
+	writeU8(os, version);
+
+	u32 num_vars = disk ? m_stringvars.size() : countNonPrivate();
+	writeU32(os, num_vars);
+	for (const auto &sv : m_stringvars) {
+		bool priv = isPrivate(sv.first);
+		if (!disk && priv)
+			continue;
+
+		os << serializeString(sv.first);
+		os << serializeLongString(sv.second);
+		writeU8(os, priv ? 1 : 0);
+	}
+}
+
+bool BlockMetadata::deSerialize(std::istream &is)
+{
+	clear();
+
+	u8 version = readU8(is);
+	if (version == 0) {
+		return false;
+	} else if (version > 1) {
```

Comment:
Assume backwards-compatibility (remove this check). This will make future changes a PITA, seriously.

Reply:
Check is removed.